Code/Framework/AzCore/AzCore/std/ranges/ranges_algorithm.h

```diff
@@ -1218,4 +1242,227 @@ namespace AZStd::ranges
         constexpr Internal::count_fn count{};
         constexpr Internal::count_if_fn count_if{};
     }
+
+    // Mutating Sequence Operations
+    // https://eel.is/c++draft/algorithms#alg.modifying.operations
+
+    // ranges::copy
+    // ranges::copy_if
+    // ranges::copy_n
+    // ranges::copy_backward
+    template<class I, class O>
+    using copy_result = in_out_result<I, O>;
+    template<class I, class O>
+    using copy_if_result = in_out_result<I, O>;
+    template<class I, class O>
+    using copy_n_result = in_out_result<I, O>;
+    template<class I1, class I2>
+    using copy_backward_result = in_out_result<I1, I2>;
+
+    namespace Internal
+    {
+        struct copy_fn
+        {
+            template<class I, class S, class O>
+            constexpr auto operator()(I first, S last, O result) const
+                -> enable_if_t<conjunction_v<
+                bool_constant<input_iterator<I>>,
+                bool_constant<sentinel_for<S, I>>,
+                bool_constant<weakly_incrementable<O>>,
+                bool_constant<indirectly_copyable<I, O>>
+                >, copy_result<I, O>>
+            {
+                for (; first != last; ++first, ++result)
+                {
+                    *result = *first;
+                }
+
+                return { AZStd::move(last), AZStd::move(result) };
```

Comment:
question: You undoubtedly know better than me but I'm just curious is it definitely necessary to use `AZStd::move` here as part of the return?

Reply:
Ah of course as they're passed in by value, makes sense, sorry should have realized that. Looks good! 🙂